src/common/components/Recaptcha/Recaptcha.tsx

```diff
@@ -89,29 +89,29 @@ export const Recaptcha = ({
         if (data?.success === true && isValidAction(data?.action)) {
           if (data.score < 0.5) {
             $log.warn(`Captcha score was below the threshold: ${data.score}`)
-            onFailure(componentInstance)
+            onFailure.call(componentInstance,componentInstance)
```

Comment:
The goal was so that controller methods like `submitOrderInternal` can use `this` like normal controller methods. It's a hidden footgun if `this` is bound to `window`. But it will be great if that won't be needed in the future.